### DISCUSSION TITLE
SQL Command Category Overview

### DISCUSSION THREAD

**Discussion by gramian** [ArcadeData/arcadedb#596] [Show and tell]
Dear Developers,

I made an overview graphic of the SQL commands by categories (DDL, DML, DQL, DCL,TCL) adapted to ArcadeDB's SQL dialect, see below. If you (conceptually) like it, it can hopefully be included into the documentation in the introduction of Section 8? Please let me know if there are errors or you prefer content-wise or presentational changes first.

While making it, I wondered why adding vertices and edges is implemented as `CREATE VERTEX` and `CREATE EDGE` respectively and not `INSERT VERTEX` and `INSERT EDGE`? I guess there are historical reasons, but would it help consistency, if `INSERT X` aliases were added for the these `CREATE X` commands? Or maybe there is a particular reason not to to do that.

<img width="827" alt="sql" src="https://user-images.githubusercontent.com/4374619/200703418-89704c3b-aed6-4912-95eb-b241a2171d14.png">

**Comment by gramian** [comment]
I updated the graphic.

**Comment by lvca** [comment]
I like it. +1 to include it in the docs.

Create vertex and edge were taken from OrientDB syntax. Not sure if make any sense or what they add to the `INSERT INTO`. I should check if there is any added value to it, otherwise change the docs to suggest users to stick with the more standard `INSERT INTO`

**Comment by gramian** [comment]
Oh, I see, I thought `CREATE VERTEX` and `CREATE EDGE` are the only way to create vertices and edges, but if I understand you correctly, I can also use `INSERT INTO` to create vertices and edges?

**Reply by lvca** [reply]
Just checked: INSERT INTO works for both documents and vertices. With edges, it's different because the syntax is more explicit about from/to vertices, but it could be still doable in the future with very small work with something like:

```sql
INSERT INTO FrriendOf SET @out = #34:23, @in = #91:2323
```

as an alternative to:

```sql
CREATE EDGE FrriendOf FROM #34:23 TO #91:2323
```
